### PR TITLE
C# Signal attribute

### PR DESCRIPTION
**Pull request by paulloz**
This is about #15004.  
Introduces a `[Signal]` attribute available for delegates. Use it like `signal` in GDScript.

### CODE REVIEW

File: modules/mono/glue/cs_files/SignalAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+namespace Godot
+{
+    [AttributeUsage(AttributeTargets.Delegate)]
+    public class SignalAttribute : Attribute
+    {
+        public SignalAttribute()
+        {
+        }
+    }
+}
```
Comment (neikeq) [inline modules/mono/glue/cs_files/SignalAttribute.cs]: Missing new line at at EOF.

**Review by neikeq — APPROVED** [review]

**Comment by neikeq** [comment]
That's fantastic, thanks!

**Comment by paulloz** [comment]
Added missing new line in [`bd38347`](https://github.com/godotengine/godot/pull/16770/commits/bd38347b165dba6d807bf4754d93f313b30c50bd).

**Comment by akien-mga** [comment]
Could you squash the commits together?

**Comment by paulloz** [comment]
@akien-mga done 😄

**Comment by maikramer** [comment]
I Cant open projects after this commit

Errors in console as follow
Console errors:
ERROR: update_godot_api_cache: Mono Cache: Member GDMonoUtils::mono_cache.class_SignalAttribute is null
   At: modules/mono/mono_gd/gd_mono_utils.cpp:205.
Mono: INITIALIZED
ERROR: has_attribute: Parameter ' p_attr_class ' is null.
   At: modules/mono/mono_gd/gd_mono_class.cpp:95.
ERROR: has_attribute: Parameter ' p_attr_class ' is null.
   At: modules/mono/mono_gd/gd_mono_method.cpp:158.

**Comment by neikeq** [comment]
@maikramer You will have to remove the `.mono` folder in your project and the `mono/solutions` folder in `user://` (`%appdata/Godot%` or `~/.local/share/godot`).

This will be fixed in the future. I will have to add some versioning other than the Godot API hash to these assemblies, and make Godot unload outdated assemblies gracefully instead of exiting.

**Comment by maikramer** [comment]
Needed to delete .mono from project to stop crashing, just leaving here because i took some time to find this solution.

**Comment by neikeq** [comment]
Hmm, weird that you didn't need to delete the other folder as well. You may experience it again next time you build.

**Comment by maikramer** [comment]
Yes, having lots of problems now, deleted mono/solutions and the game started, but i lost all my [Export] variables. At least i'm with time here.

**Comment by neikeq** [comment]
That's probably #15371

**Pull request merged**